examples/credit_loan_decisions.py

```diff
@@ -0,0 +1,1001 @@
+np.random.seed(rand_seed)
+
+# %% 
+# Fairness considerations of credit loan decisions
+# ------------------------------------------------
+
+# %% 
+# Fairness and credit lending in the US
+# -----------------------------------------
+
+# %% 
+# In 2019, Apple received backlash on social media after its newly launched _Apple Card_ product appeared to `offer higher credit limits to men compared to women <https://edition.cnn.com/2019/11/12/business/apple-card-gender-bias/index.html>`_. In multiple cases, married couples found the husband received a credit limit that was 10-20x higher than the wife's even when the couple had joint assets.
+
+# %% 
+# From a regulatory perspective, financial institutions that operate within the United States are subject to _legal regulations_ prohibiting discrimination on the `basis of race, gender, or other protected classes <https://www.govinfo.gov/content/pkg/USCODE-2011-title15/html/USCODE-2011-title15-chap41-subchapIV.htm>`_. With the increasing prevalence of automated decision-systems in the financial lending space, experts have raised concerns about whether these systems could exacerabate existing inequalities in financial lending.
+
+# %% 
+# Although the two concepts are intertwined, algorithmic fairness is not the same concept as anti-discrimination law. An AI system can comply with anti-discrimination law while exhibiting fairness-related concerns. On the other hand, some fairness interventions may be illegal under anti-discrimination law. `Xiang and Raji  (2019) <https://arxiv.org/abs/1912.00761>`_` discuss the compatibilities and disconnects between anti-discrimination law and algorithmic notions of fairness. In this case study, we focus on fairness in financial services rather than compliance with financial anti-discrimination regulations.
+
+# %% 
+# *EY* case study
+# ==================
+
+# %% 
+# In this case study, we aim to replicate the work done in a `white paper <https://www.microsoft.com/en-us/research/uploads/prod/2020/09/Fairlearn-EY_WhitePaper-2020-09-22.pdf>`_, co-authored by _Microsoft_ and _EY_, on mitigating gender-related performance disparities in financial lending decisions. In their analysis, _Microsoft_ and _EY_ demonstrated how _Fairlearn_ could be used to measure and mitigate unfairness in the loan adjudication process.
+
+# %% 
+# Using a dataset of credit loan outcomes (whether an individual defaulted on a credit loan), we train a fairness-unaware model to predict the likelihood an individual will default on a given loan. We use the *Fairlearn* toolkit for assessing the fairness of our model, according to several metrics. Finally, we perform two unfairness mitigation strategies on our model and compare the results to our original model.
+
+# %% 
+# Because the dataset used in the white paper is not publicly available, we will introduce a semi-synthetic feature into an existing publicly available dataset to replicate the outcome disparity found in the original dataset.
```

Comment:
As long as there are no code, these can be in a single cell, we don't need to start a new cell for each paragraph with `# %%`. It'd also be nice to wrap the long lines to shorter ones, and you can use `alt+q` in emacs, or the same key in VSCode if you install the *Rewrap* extension to wrap the lines.